packages/core/js-client/src/api.ts

```diff
@@ -33,38 +42,47 @@ export const isFluencePeer = (fluencePeerCandidate: unknown): fluencePeerCandida
  * @param script - air script with function execution logic generated by the Aqua compiler
  */
 export const v5_callFunction = async (
-    rawFnArgs: Array<any>,
-    def: FunctionCallDef,
-    script: string,
+  rawFnArgs: unknown[],
+  def: FunctionCallDef,
+  script: string,
 ): Promise<unknown> => {
-    const { args, client: peer, config } = await extractFunctionArgs(rawFnArgs, def);
-    
-    return callAquaFunction({
-        args,
-        def,
-        script,
-        config: config || {},
-        peer: peer,
-    });
+  const { args, client: peer, config } = extractFunctionArgs(rawFnArgs, def);
+
+  return callAquaFunction({
+    args,
+    def,
+    script,
+    config: config,
+    peer: peer,
+  });
 };
 
 /**
  * Convenience function to support Aqua `service` generation backend
  * The compiler only need to generate a call the function and provide the corresponding definitions and the air script
  * @param args - raw arguments passed by user to the generated function
+ * TODO: dont forget to add jsdoc for new arg
  * @param def - service definition generated by the Aqua compiler
  */
-export const v5_registerService = async (args: any[], def: ServiceDef): Promise<unknown> => {
-    const { peer, service, serviceId } = await extractServiceArgs(args, def.defaultServiceId);
-    
-    return registerService({
-        def,
-        service,
-        serviceId,
-        peer,
-    });
+export const v5_registerService = (args: unknown[], def: ServiceDef): void => {
+  // TODO: Support this in aqua-to-js package
+  // eslint-disable-next-line @typescript-eslint/consistent-type-assertions
```

Comment:
also I just now realised that aqua-to-js still uses overloads even though there is no reason for that. Instead of overloads just union can be used e.g.

function registerSomeService(...args: [service: SomeServiceDef] | [serviceId: string, service: SomeServiceDef] | [peer: IFluenceClient$$, service: SomeServiceDef] | [peer: IFluenceClient$$, serviceId: string, service: SomeServiceDef])

Reply:
I hate argument ordering here and that hard-to-read union type

Reply:
But changing something here means api will drastically changes. Need to discuss this first

Reply:
with what I suggest you will still have argument ordering exactly like before. It's just that it will be implemented in types using unions instead of overloads which is less syntax, better typescript error messages while the api remains practically the same

here there is no need for function overloads because no mater what arguments function accepts it returns the same value in all the cases. So literally no value at all to have overloads instead of simple unions

Feel free to create a separate task for that, but I think this should be done for simplification and better error messages => better user experience

Reply:
Sounds reasonable

Reply:
Here is the task for it
https://linear.app/fluence/issue/DXJ-498/change-argument-types-in-generated-js